atlas.go

```diff
@@ -35,6 +35,11 @@ func (n *Node) IsRoot() bool {
 	return n.root == n
 }
 
+// GetNodeData returns the previous node and the key and value of the current one.
+func (n *Node) GetNodeData() (*Node, string, string) {
```

Comment:
I would argue this is a rare case in which named returns make sense ;) 

```suggestion
func (n *Node) GetNodeData() (prev *Node, key string, value string) {
```

I don't really have better name if we want all three at the same time. Separating themi in 2 methods can be

`Prev`/`Parent`
and 
`KeyValue`/`Data` !?!


I would Also prever to drop the `Get` from thsi method if possible ;)

Reply:
When iterating over them, users will probably almost always want all 3, and they can use `_` if they don't. For example, here's how encoding them in JSON worked out in k6:
```go
func (ts *TagSet) MarshalEasyJSON(w *jwriter.Writer) {
	w.RawByte('{')
	first := true

	n := (*atlas.Node)(ts)
	for !n.IsRoot() {
		prev, key, value := n.GetNodeData()
		if first {
			first = false
		} else {
			w.RawByte(',')
		}
		w.String(key)
		w.RawByte(':')
		w.String(value)
		n = prev
	}
	w.RawByte('}')
}
```

Which seems fairly readable? :+1: on removing the `Get` and the named returns though, so how about this:

```suggestion
func (n *Node) NodeData() (prev *Node, key string, value string) {
```

:question:

Reply:
:+1: 

maybe add an example with this (simplified) either as Example or just as part of the docs :thinking:

Reply:
Resolved by the [force-push](https://github.com/mstoykov/atlas/compare/1ab3558876aaed3042eda88d0bc526d0c075027f..75bb75c1ca5bb17aaee0fa2abbd1aab9b1e74d10)